synfig-studio/src/gui/widgets/widget_timeslider.cpp

```diff
@@ -349,21 +388,65 @@ bool
 Widget_Timeslider::on_button_release_event(GdkEventButton *event){
 	SYNFIG_EXCEPTION_GUARD_BEGIN()
 	lastx = (double)event->x;
+	move_lower_bound_button = false;
+	move_upper_bound_button = false;
 	return event->button == 1 || event->button == 2;
 	SYNFIG_EXCEPTION_GUARD_END_BOOL(true)
 }
 
+void
+Widget_Timeslider::get_bounds_rectangle_dimensions(double& x0, double& x1, double& w ,bool lower)
+{
+	if(lower){
+		 x0 = time_plot_data->get_double_pixel_t_coord(time_plot_data->lower_ex);
+		 x1 = time_plot_data->get_double_pixel_t_coord(time_plot_data->time_model->get_play_bounds_lower());
+		 w = x1 - x0;
+	}
+	else {
+		 x0= time_plot_data->get_double_pixel_t_coord(time_plot_data->time_model->get_play_bounds_upper());
+		 x1= time_plot_data->get_double_pixel_t_coord(time_plot_data->upper_ex);
+		 w= x1 - x0;
+	}
+}
+
 bool
 Widget_Timeslider::on_motion_notify_event(GdkEventMotion* event) //for dragging
 {
 	SYNFIG_EXCEPTION_GUARD_BEGIN()
+
+	double current= time_plot_data->get_double_pixel_t_coord(time_plot_data->get_t_from_pixel_coord(event->x));
+	double x0,x1,w,z0,z1,width;
+	get_bounds_rectangle_dimensions(x0,x1,w,true);
+	get_bounds_rectangle_dimensions(z0,z1,width,false);
+
+	if((((current <= (x0+w))  && (current >= (x0 + w - 4)) && !move_lower_bound_button) || (move_lower_bound_button))){
+		if((get_window()->get_cursor() != bounds_cursor))
+		   get_window()->set_cursor(bounds_cursor);
+	}
+	else if((((current <= (z0 + 4))  && (current >= (z0)) && !move_upper_bound_button) || (move_upper_bound_button))){
+		if((get_window()->get_cursor() != bounds_cursor))
+		   get_window()->set_cursor(bounds_cursor);
+	}
+	else {
+		if((get_window()->get_cursor() != default_cursor))
```

Comment:
I'm not sure if it is necessary to check it before calling `set_cursor`

Reply:
yes, it seems redundant will remove.

Reply:
@rodolforg  actually now I remember why I did that it was to avoid the constant setting of the cursor, as you see this is in the motion notify event.

Reply:
The set-cursor method does not check it then? Oh ok

Reply:
No in fact you're correct, loooking at the source [code](https://gitlab.gnome.org/GNOME/gtk/-/blob/main/gtk/gtkwidget.c#L12144)  line 12165 to be exact 
supposedly `g_set_object()` returns if the value had been changed or not and if not changed the `set_cursor()` method returns. so In fact my extra check here seems redundant will remove now.